src/main/java/software/amazon/smithy/lsp/SmithyInterface.java

```diff
@@ -37,4 +41,34 @@ private SmithyInterface() {
       return Either.forLeft(e);
     }
   }
+
+  public static Either<Exception, ValidatedResult<Model>> readModel(File path, Set<String> externalJars) {
+
+    try {
+      ModelAssembler assembler = Model.assembler().discoverModels();
```

Comment:
I'm not sure we should allow for model discovery if the LSP owns the resolution of artifacts